src/cli/run.ts

```diff
@@ -36,6 +37,7 @@ export const run = async (init: d.CliInitOptions) => {
     }
 
     if (task === 'help' || flags.help) {
+      // TODO(NOW): As we expand this type with more strict values, this temp config will grow :-(
       await taskHelp({ flags: { task: 'help', args }, outputTargets: [] }, logger, sys);
```

Comment:
One way I toyed with last night was changing the signature of `taskHelp`, such that it doesn't need a config object (it only uses flags ATM):
```diff
-export const taskHelp = async (config: d.InternalStrictConfig, logger: d.Logger, sys?: d.CompilerSystem) => {
+export const taskHelp = async (flags: d.ConfigFlags, logger: d.Logger, sys?: d.CompilerSystem) => {
```

Reply:
I kind of like the approach of changing it to just take a `ConfigFlags` argument. Then the contract between the function and the rest of the codebase as documented in the type signature is a bit more minimal and is therefore more representative of the actual dependencies that the function has i.e. which things it actually is going to poke at. 

An alternative to the manually-created minimal config thing could be that we add a function which, for the sake of example, is called `createMinimalStrictConfig`, and could do something like this:

```ts
const createMinimalStrictConfig = (starter: Loose<Config>): InternalStrictConfig => {
  const config: InternalStrictConfig = {
    flags: createFlags(),
    // whatever other fields we need to instantiate
    ...starter
  }
  return config
}
```

or something like that. Then if we end up with a few similar configs that we need to create in-line to support use-cases like the 'help' flag here we'd just have a call to `createMinimalStrictConfig`, and adding a new required field to `InternalStrictConfig` would only require a change in one place.

Reply:
> I kind of like the approach of changing it to just take a ConfigFlags argument. Then the contract between the function and the rest of the codebase as documented in the type signature is a bit more minimal and is therefore more representative of the actual dependencies that the function has i.e. which things it actually is going to poke at.

Agreed.  I ended up doing something similar on Friday (I didn't push it up) where I did just that.  Turns out a lot of the `config` usage here is just like you said - it's a vehicle for `flags` and...nothing else. I think that's the way to go, especially since `--help` doesn't (and IMO shouldn't) require a config

Reply:
yeah agreed - I'd like to have some assurance that a user who installs Stencil and has no config set up will still see a help message without being blocked by some config validation issue or something